src/config/settings.ts

```diff
@@ -68,6 +68,7 @@ export const defaultSettings = {
     showMuted:   false,
     mutedChecks: {} as Record<string, boolean>,
   },
+  preferences: { currentNavItem: 'Application' },
```

Comment:
I'm contemplating if it makes sense to to write the `currentNavItem` to our config and I'm leaning toward no.. 

I'm thinking of the use case where I open Preferences, change the container engine, close Rancher Desktop, and re-open Rancher Desktop some time later. IMO it subverts expectations to have Container Engine as the selected item in Preferences after closing and re-opening. I would expect the first menu item to be selected.

It might suffice to just store the `currentNavItem` in the preferences store and lose this information when closing and reopening Rancher Desktop.

@manuelecarlini @mook-as how do you feel about this?

Reply:
NOTE: I'm also wondering if the linked issue might be prescribing an issue where we might be lacking some useful information elsewhere in Rancher Desktop. i.e. does it make sense to require users to open preferences in order to see the selected K8S version, are we missing an opportunity to expose this information in a more readily accessible location? This is probably something that we can design for later.

Reply:
@rak-phillip I also think it is better to restore the default page once the user decide to restart the Application. I have prepared a new [commit](https://github.com/rancher-sandbox/rancher-desktop/pull/3010/commits/e0997ba861390536443f6680cad05c13a2bec110) in the case we decide to do so.

Reply:
I know I'm not involved with this PR, but I can't help but comment. Remembering tabs is not related to settings. We should not store the last accessed preferences tab in our settings object, nor should we store it in settings.json. I would be very surprised if I came upon a key like this in our config... and we want to avoid being surprised, because that's how bugs are made.

On Linux, the XDG spec has `XDG_STATE_HOME` and `XDG_CACHE_HOME` for things like this. Config goes in `XDG_CONFIG_HOME`. I'm sure that there are equivalents for other operating systems.

So I agree with you guys, it would be better to store it in the preferences store. Or if we want to persist it across restarts, in `XDG_STATE_HOME` or `XDG_CACHE_HOME` and their equivalents on macOS and Windows.

Reply:
I don't think it should be persisted across restarts, only as long as the app remains open.

As @adamkpickering points out, these are not settings, but just active application state. Similar to how we do not persist the size and position of the application window on the desktop.

I remember a time when applications (on Windows) used to do that; is that still a thing?

Reply:
It sounds pretty unanimous that we don't believe the settings object to be the correct location to store the `currentNavItem`, so that begs the question of "where do we store it?"

I spoke briefly with @mook-as and it sounds like we `transientSettings` might be the correct mechanism for storing data like this. If we were to use this, we would need to expose `transientSettings` to both `main` and `renderer` processes by either:

1. Add `transientSettings` to `settings` and make sure that we never write `transientSettings` to disk
2. Expose a new api endpoint for `transientSettings` that just reads and writes values in memory

Reply:
I will try to make a recap:
We discarded the possibility to keep the latest tab across application restarts, IMHO this means there is no point in persisting that information in the file system.
So if you agree, I would proceed as @rak-phillip suggested, to store data in memory using `transientSettings`.